### PR TITLE
Added github dependabot connector

### PR DESCRIPTION
**Pull request by imishraSecurview**

### CODE REVIEW

**Comment by lindambrown** [comment]
@imishraSecurview please address lint error. You should be pointing to rule file in the toolkit director to make sure your code editor is catching all the error based on Kenna standards. Thanks!

File: tasks/connectors/github_dependabot/github_dependabot.rb [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        @kenna_connector_id = @options[:kenna_connector_id]
+
+        repos = GithubDependabot.security_advisories.original_hash["data"]["organization"]["repositories"]["nodes"] # get the response from graphql query
+
+        repo_map = repos.each_with_object({}) do |repo, h|
+          advisories = repo["vulnerabilityAlerts"]["nodes"].map { |alert| alert["securityAdvisory"] }
+          h[repo["name"]] = advisories.reject { |ad| ad["identifiers"].last.value?("GHSA") }
+        end
+
+        kdi_hash = {
```
Comment (lindambrown) [inline tasks/connectors/github_dependabot/github_dependabot.rb:116]: please review and use the help methods to complete the functions from 116 to 150

create_kdi_asset_vuln(asset, vulnerability_alerts_for(repo))
create_kdi_vuln_def(vulnerability_definitions(repo_map))

and then 

kdi_upload(@output_directory, "sample_scan_#{scan_id}_report_#{pos}.json", @kenna_connector_id, @kenna_api_host, @kenna_api_key, @skip_autoclose, @retries, @kdi_version)

let me know if you have any questions.

File: tasks/connectors/github_dependabot/github_dependabot_readme.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+>>https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token
+
+
+
+## Complete list of Options:
+
+| Option | Required | Description | default |
+| --- | --- | --- | --- |
+| github_access_token | true | Github access token | n/a |
+| atp_api_host | false | url to retrieve Defender hosts and vulns | https://api.securitycenter.microsoft.com |
```
Comment (lindambrown) [inline tasks/connectors/github_dependabot/github_dependabot_readme.md:21]: couple of old references to ATP here

File: toolkit.rb [hunk 3]
```diff
@@ -31,6 +29,11 @@
   args[arg_name] = arg_value
 end
 
+# constant set for graphql which is used in github dependabot connector
```
Comment (lindambrown) [inline toolkit.rb:32]: no change should be needed to this file

Comment (lindambrown) [reply toolkit.rb:32]: @imishraSecurview

File: tasks/connectors/github_dependabot/github_dependabot_readme.md (hunk 2 above)
Comment (lindambrown) [reply tasks/connectors/github_dependabot/github_dependabot_readme.md:21]: @imishraSecurview

File: tasks/connectors/github_dependabot/github_dependabot.rb (hunk 1 above)
Comment (lindambrown) [reply tasks/connectors/github_dependabot/github_dependabot.rb:116]: @imishraSecurview

File: tasks/connectors/github_dependabot/github_dependabot.rb [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+      Schema = GraphQL::Client.load_schema(http)
+
+      Client = GraphQL::Client.new(schema: Schema, execute: http)
+
+      # Graphql query
+
+      SECURITY_ADVISORY_QUERY = Client.parse <<-'GRAPHQL'
+        query {
+          organization(login: "KennaSecurity") {
```
Comment (resnickio) [inline tasks/connectors/github_dependabot/github_dependabot.rb:59]: "KennaSecurity" should be modified to be a variable that can be passed through on the command line. As written, all runs only produce results for the KennaSecurity organization and assets are empty when the personal access token does not have permissions for KennaSecurity Dependabot alerts

File: toolkit.rb (hunk 3 above)
Comment (imishraSecurview) [reply toolkit.rb:32]: @lindambrown here i am looking for github_access_token, that psss from command line. the graphql configuration defined with constant, so when i run toolkit command from cli. the graphql configuration execute before run(opts) method. hence i have added github_access_token constant in toolkit.rb .

**Review by lindambrown — APPROVED** [review]

**Pull request merged**